src/babel/transformation/file/options.json

```diff
@@ -21,8 +21,8 @@
   },
 
   "moduleId": {
-    "description": "specify a custom name for module ids",
-    "type": "string"
+    "description": "specify a custom name for module ids, or a function to transform the default name",
+    "type": ["string", "function"]
```

Comment:
Okay, so `getModuleName`? What should the behavior be if `getModuleName` is provided as well as a `moduleId`? Does `moduleId` become noop?